e2e/ci-e2e.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+
+################################################################################
+#
+# This script is meant to be run from the root of this project
+#
+################################################################################
+
+set -euo pipefail
```

Comment:
I would recommend adding the `-x` flag in addition to the rest since it will print all the executed commands which helps to debug now and in the future.